sycl/include/sycl/detail/defines_elementary.hpp

```diff
@@ -39,7 +39,7 @@
 #ifndef __SYCL_DEPRECATED
 // The deprecated attribute is not supported in some situations(e.g. namespace)
 // in C++14 mode
-#if !defined(SYCL2020_DISABLE_DEPRECATION_WARNINGS) && __cplusplus >= 201703L
+#if !defined(SYCL_DISABLE_DEPRECATION_WARNINGS) && __cplusplus >= 201703L
```

Comment:
This doesn't seem right to me. The `SYCL2020_DISABLE_DEPRECATION_WARNINGS` macro is in the source base in about 20 places. It's both documented and prepped in certain .cpp files.  

The symbol `SYCL_DISABLE_DEPRECATION_WARNINGS` may have once existed, but it exists solely in comments now. There are no docs for it, and no .cpp file in the source base preps it.     

Perhaps it was once the published solution to stop deprecation warnings, but was unintentionally dropped?  I don't know.  But in any case, wouldn't the suggested change will break the intended use of the  `SYCL2020_DISABLE_DEPRECATION_WARNINGS` macro (at least in the C++17 only sections)?  Maybe we should `||` both symbols in this clause.

Reply:
Not sure if it matters, but AFAIK the `__cplusplus >= 201703L` piece is guaranteed now (we `static_assert` it).